src/frontend/js/index.js

```diff
@@ -845,6 +720,14 @@ function meta2options(metadata, selected) {
         .map(([k, v]) => m('option', {value: k, selected: selected === k}, v.name || k))
 }
 
+function scenarios_with(meta, variable) {
+    const v = meta.od_matrices[variable]
+    if (v === undefined)
+        return [];
+    else
+        return Object.fromEntries(Object.entries(meta.scenarios).filter(([scen, _]) => R.contains(scen, v.scenarios_with)))
```

Comment:
```suggestion
        return R.pickBy((_, key) => R.contains(key, v.scenarios_with), meta.scenarios)
```

Pretty sure this does the same thing. Untested